src/helpers/DelegationMetaSwapAdapter.sol

```diff
@@ -341,6 +365,42 @@ contract DelegationMetaSwapAdapter is ExecutionHelper, Ownable2Step {
         emit SentTokens(_token, _recipient, _amount);
     }
 
+    /**
+     * @dev Validates that the tokens are whitelisted or not based on the _useTokenWhitelist flag.
+     * @dev Adds the argsCheckEnforcer args to later validate if the token whitelist must be have been used or not.
+     * @param _tokenFrom The input token of the swap.
+     * @param _tokenTo The output token of the swap.
+     * @param _delegations The delegation chain; the last delegation must include the ArgsEqualityCheckEnforcer.
+     * @param _useTokenWhitelist Flag indicating whether token whitelist checks should be enforced.
+     */
+    function _validateTokens(
+        IERC20 _tokenFrom,
+        IERC20 _tokenTo,
+        Delegation[] memory _delegations,
+        bool _useTokenWhitelist
+    )
+        private
+        view
+    {
+        // The Args Enforcer must be the first caveat in the root delegation
+        uint256 lastIndex_ = _delegations.length - 1;
+        if (
+            _delegations[lastIndex_].caveats.length == 0
+                || _delegations[lastIndex_].caveats[0].enforcer != argsEqualityCheckEnforcer
+        ) {
+            revert MissingArgsEqualityCheckEnforcer();
+        }
+
+        // The args are set by this contract depending on the useTokenWhitelist flag
+        if (_useTokenWhitelist) {
+            if (!isTokenAllowed[_tokenFrom]) revert TokenFromIsNotAllowed(_tokenFrom);
+            if (!isTokenAllowed[_tokenTo]) revert TokenToIsNotAllowed(_tokenTo);
+            _delegations[lastIndex_].caveats[0].args = abi.encode("Enforce Token Whitelist");
```

Comment:
lets make this less error prone but still readable.

Maybe something like "Token-Whitelist-Enforced" and "Token-Whitelist-Not-Enforced" ? The spaces may cause a hard to find error.

Maybe we could even create a custom error and use it as a selector `error TokenWhitelistEnforced()` and then gator sdk could do something like `_delegations[lastIndex_].caveats[0].args = DelegationManager.TokenWhitelistEnforced.selector`